internal/healthchecks/healthchecks.go

```diff
@@ -56,30 +88,27 @@ func (r HealthCheckResult) String() string {
 	return singleErrorResultMessage(r.Err, r.Name)
 }
 
-func LogHealthCheckResults(healthCheckResults []HealthCheckResult, infoLogger func(string), errorLogger func(string)) {
+func LogHealthCheckResults(healthCheckResults []HealthCheckResult, infoLogger func([]string), errorLogger func([]string)) {
 	for _, result := range healthCheckResults {
-		if result.Err != nil {
-			errorLogger(result.String())
-		} else {
-			infoLogger(result.String())
-		}
+		errorLogger(result.ErrorMessages())
+		infoLogger(result.InfoMessages())
 	}
 }
 
-func CreateHealthChecksLogger(logDir string) (logger *log.Logger, closer func()) {
+func CreateHealthChecksLogger(logDir string) (logger *logs.StructuredLogger, closer func()) {
 	path := filepath.Join(logDir, healthChecksLogFile)
 	// Make sure the directory exists before writing the file.
 	if err := os.MkdirAll(logDir, 0755); err != nil {
 		log.Printf("failed to create directory for %q: %v", path, err)
-		return log.Default(), func() {}
+		return logs.Default(), func() {}
 	}
 	file, err := os.OpenFile(path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
 	if err != nil {
 		log.Printf("failed to open health checks log file %q: %v", path, err)
-		return log.Default(), func() {}
+		return logs.Default(), func() {}
 	}
 
-	return log.New(file, "", log.Ldate|log.Ltime|log.Lshortfile), func() { file.Close() }
+	return logs.New(path), func() { file.Close() }
```

Comment:
Since we are not using the opened `file` anymore to create the logger, should we close ( `file.Close()`) it before ending this function ? Is it needed to open it any more ? 

Mostly a comment to address on how to handle the opened files and where to close them.